include/deal.II/fe/fe_enriched.h

```diff
@@ -691,6 +696,311 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
+   * std::shared_ptr< hp::FECollection<dim> > fe_collection;
+   *
+   * Triangulation<dim>  triangulation;
+   * hp::DoFHandler<dim> dof_handler(triangulation);
+   *
+   * static ColorEnriched::Helper<dim> FE_helper(fe_base,
+   *                                             fe_enriched,
+   *                                             predicates,
+   *                                             enrichments);
+   * fe_collection = FE_helper.get_fe_collection(dof_handler);
+   * @endcode
+   */
+  template <int dim, int spacedim = dim>
+  struct Helper
+  {
+    /**
+     * Constructor for Helper class.
+     *
+     * @param fe_base A base FiniteElement
+     * @param fe_enriched An enriched FiniteElement
+     * @param predicates std::vector of predicates defining the sub-domains.
+     * <code>@p predicates[i]</code> returns true for a cell if it
+     * belongs to a sub-domain with index (i).
+     * @param enrichments std::vector of enrichment functions
+     */
+    Helper(const FE_Q<dim, spacedim> &                             fe_base,
+           const FE_Q<dim, spacedim> &                             fe_enriched,
+           const std::vector<predicate_function<dim, spacedim>> &  predicates,
+           const std::vector<std::shared_ptr<Function<spacedim>>> &enrichments);
+
+    /**
+     * Prepares an hp::DoFHandler object. The active FE indices of
+     * mesh cells are initialized to work with
+     * ColorEnriched::Helper<dim,spacedim>::fe_collection.
+     *
+     * Returns an hp::FECollection object.
+     *
+     * @param dof_handler an hp::DoFHandler object
+     * @return shared pointer to hp::FECollection, a collection of
+     * finite elements needed by @p dof_handler.
+     */
+    std::shared_ptr<hp::FECollection<dim, spacedim>>
+    get_fe_collection(hp::DoFHandler<dim, spacedim> &dof_handler);
+
+  private:
+    /**
+     * Contains a shared pointer to a collection of FiniteElement objects
+     */
+    std::shared_ptr<hp::FECollection<dim, spacedim>> fe_collection;
+
+    /**
+     * A base FiniteElement used for constructing FE_Enriched
+     * object required by ColorEnriched::Helper<dim,spacedim>::fe_collection.
+     */
+    const FE_Q<dim, spacedim> &fe_base;
+
+    /**
+     * An enriched FiniteElement used for constructing FE_Enriched
+     * object required by ColorEnriched::Helper<dim,spacedim>::fe_collection.
+     */
+    const FE_Q<dim, spacedim> &fe_enriched;
+
+    /**
+     * A finite element with zero degrees of freedom used for
+     * constructing FE_Enriched object required by
+     * ColorEnriched::Helper<dim,spacedim>::fe_collection
+     */
+    const FE_Nothing<dim, spacedim> fe_nothing;
+
+    /**
+     * std::vector of predicates defining the sub-domains.
+     * <code>@p predicates[i]</code> returns true for a cell if it
+     * belongs to a sub-domain with index (i).
+     */
+    std::vector<predicate_function<dim, spacedim>> predicates;
```

Comment:
i think that can be `const std::vector<predicate_function<dim, spacedim>>&`?

Reply:
Thanks. I think that would be better. I will try.